src/layer/Popup.js

```diff
@@ -401,67 +401,56 @@ Layer.include({
 	// @method unbindPopup(): this
 	// Removes the popup previously bound with `bindPopup`.
 	unbindPopup: function () {
-		if (this._popup) {
-			this.off({
-				click: this._openPopup,
-				keypress: this._onKeyPress,
-				remove: this.closePopup,
-				move: this._movePopup
-			});
-			this._popupHandlersAdded = false;
-			this._popup = null;
-		}
+		this.off({
+			click: this._openPopup,
+			keypress: this._onKeyPress,
+			remove: this.closePopup,
+			move: this._movePopup
+		});
+		this._popupHandlersAdded = false;
+		this._popup = null;
+
 		return this;
 	},
 
 	// @method openPopup(latlng?: LatLng): this
 	// Opens the bound popup at the specified `latlng` or at the default popup anchor if no `latlng` is passed.
 	openPopup: function (layer, latlng) {
-		if (this._popup && this._map) {
-			latlng = this._popup._prepareOpen(this, layer, latlng);
+		latlng = this._popup._prepareOpen(this, layer, latlng);
 
-			// open the popup on the map
-			this._map.openPopup(this._popup, latlng);
-		}
+		// open the popup on the map
+		this._map.openPopup(this._popup, latlng);
 
 		return this;
 	},
 
 	// @method closePopup(): this
 	// Closes the popup bound to this layer if it is open.
 	closePopup: function () {
-		if (this._popup) {
-			this._popup._close();
-		}
-		return this;
+		return this._popup._close();

```

Comment:
I think we should keep checking `if (this._popup) {` because logics which want to make sure that popups get closed are breaking now.